### PR TITLE
Query payments example

### PR DESCRIPTION
**Pull request by d0rich**

### CODE REVIEW

File: content/4.examples/query-payment-states.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
+# Query payment states
+
+Corda stores the state of a payment in the vault of the node that initiated the payment. So you can access the state of a payment by querying the vault.
```
Comment (lingvolab) [inline content/4.examples/query-payment-states.md:3]:
```suggestion
Corda stores the state of a payment in the vault of the node that initiated the payment, so you can access the state of a payment by querying the vault.
```

File: content/4.examples/query-payment-states.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    + getPayment()
+    + queryPayments()
+  }
+
+  MyPaymentService ..> GevamuFacade
+```
+
+## Payment structure
+
+Structure of the [`Payment`](https://gevamu.github.io/corda-payments-sdk/payments-contracts/com.gevamu.corda.states/-payment/index.html) is as follows:
```
Comment (lingvolab) [inline content/4.examples/query-payment-states.md:20]:
```suggestion
Structure of the [`Payment`](https://gevamu.github.io/corda-payments-sdk/payments-contracts/com.gevamu.corda.states/-payment/index.html) state is as follows:
```

File: content/4.examples/query-payment-states.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+::notice{info}
+  Pay attention, that full payment instruction is stored as Corda [`Attachment`](https://docs.r3.com/en/api-ref/corda/4.8/open-source/javadoc/net/corda/core/contracts/Attachment.html). So you need to download it and deserialize it to get all payment details.
```
Comment (lingvolab) [inline content/4.examples/query-payment-states.md:23]:
```suggestion
 Note that the full payment instruction is stored as a Corda [`Attachment`](https://docs.r3.com/en/api-ref/corda/4.8/open-source/javadoc/net/corda/core/contracts/Attachment.html). To get all payment details, download the attachment and deserialize it.
```

File: content/4.examples/query-payment-states.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    + open(): InputStream
+  }
+
+  Payment o-- PaymentStatus
+  Payment .. Attachment: paymentInstructionId --> id
+```
+
+## About vault queries
+
+Every Corda node have a vault that stores all states that are created by the node. Our interest is in the `Payment` states.
```
Comment (lingvolab) [inline content/4.examples/query-payment-states.md:61]:
```suggestion
Every Corda node has a vault that stores all the states created by the node (in our case, the `Payment` states).
```

File: content/4.examples/query-payment-states.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+Corda provides a powerful query language to search for states in the vault. You can find more information about it in the [Writing vault queries](https://docs.r3.com/en/platform/corda/4.7/enterprise/cordapps/api-vault-query.html) section of the documentation.
```
Comment (lingvolab) [inline content/4.examples/query-payment-states.md:63]:
```suggestion
Corda provides a powerful query language to search for states in the vault – learn more about [Writing vault queries](https://docs.r3.com/en/platform/corda/4.7/enterprise/cordapps/api-vault-query.html) in Corda documentation.
```

File: content/4.examples/query-payment-states.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+::notice{warning}
+  Payment attachments are stored as files. So you can't use vault queries to search by payment details stored in attachments.
```
Comment (lingvolab) [inline content/4.examples/query-payment-states.md:66]:
```suggestion
However, you cannot use vault queries to search by payment details inside Payment attachments, as they are stored as files.
```

File: content/4.examples/query-payment-states.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+::
+
+In order to create queries for Gevamu payments you need following imports:
```
Comment (lingvolab) [inline content/4.examples/query-payment-states.md:69]:
```suggestion
To create queries for the Gevamu payments, use the following imports:
```

File: content/4.examples/query-payment-states.md [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+// Corda imports
+import net.corda.core.node.services.vault.QueryCriteria
+// Gevamu SDK imports
+import com.gevamu.corda.schema.PaymentSchemaV1
+import com.gevamu.corda.states.Payment
+```
+
+## Get single payment
+
+In order to find unique payment it is necessary to search by `uniquePaymentId` field. That's why `getPayment` method accepts `uniquePaymentId` as a parameter.
```
Comment (lingvolab) [inline content/4.examples/query-payment-states.md:81]:
```suggestion
To find a unique payment, search by the `uniquePaymentId` field, accepted by the `getPayment` method as a parameter.
```

File: content/4.examples/query-payment-states.md [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+        )
+        // Query vault and return single payment state
+        return serviceHub.vaultService.queryBy<Payment>(criteria).states.singleOrNull()
+    }
+}
+```
+
+## Query payments
+
+Criterias for searching payments can be unique for different businesses. So next example shows generic method for querying payments.
```
Comment (lingvolab) [inline content/4.examples/query-payment-states.md:104]:
```suggestion
Search criteria for payments can be unique to particular businesses. The following example shows a generic method for querying payments.
```

File: content/4.examples/query-payment-states.md [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+    fun queryPayments(criteria: QueryCriteria): List<StateAndRef<Payment>> {
+        // Query vault and return list of payment states
+        return serviceHub.vaultService.queryBy<Payment>(criteria).states
+    }
+}
+```
+
+### Query payments by date range
+
+Querying objects by date range is a common task for many businesses. So following example shows how to implement this functionality.
```
Comment (lingvolab) [inline content/4.examples/query-payment-states.md:120]:
```suggestion
Querying objects by a date range is a common task for many businesses. The following example shows how to implement this functionality.
```

**Review by lingvolab — CHANGES_REQUESTED** [review]

File: content/4.examples/query-payment-states.md (hunk 6 above)
Comment (ASitnikov) [reply content/4.examples/query-payment-states.md:66]: Gevamu SDK doesn't provide ready to use solution for querying by payments details stored in the attachment. However, you can create additional tables with the information needed for your business needs and use them in vault queries.

@lingvolab please proofread it

File: content/4.examples/query-payment-states.md [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+
+```kotlin
+// Corda imports
+import net.corda.core.node.services.vault.QueryCriteria
+// Gevamu SDK imports
+import com.gevamu.corda.schema.PaymentSchemaV1
+import com.gevamu.corda.states.Payment
+```
+
+## Get single payment
```
Comment (ASitnikov) [inline content/4.examples/query-payment-states.md:79]:
```suggestion
## Get a single payment
```

**Review by ASitnikov — CHANGES_REQUESTED** [review]

File: content/4.examples/query-payment-states.md (hunk 3 above)
Comment (d0rich) [reply content/4.examples/query-payment-states.md:23]: I would like to save casual relationship between sentences. Consequences of storing instructions as attachment are the reason, why it is inside `notice`

File: content/4.examples/query-payment-states.md (hunk 6 above)
Comment (lingvolab) [reply content/4.examples/query-payment-states.md:66]:
```suggestion
  Gevamu SDK doesn't provide any ready-to-use solution for querying by payments details stored in the attachment. However, you can create additional tables with the information needed for your business needs and use them in vault queries.
```

File: content/4.examples/query-payment-states.md [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+  class Attachment {
+    <<interface>>
+    + id: SecureHash
+    + open(): InputStream
+  }
+
+  Payment o-- PaymentStatus
+  Payment .. Attachment: paymentInstructionId --> id
+```
+
```
Comment (lingvolab) [inline content/4.examples/query-payment-states.md:58]:
```suggestion
::notice{info}
  Note that the full payment instruction is stored as a Corda [`Attachment`](https://docs.r3.com/en/api-ref/corda/4.8/open-source/javadoc/net/corda/core/contracts/Attachment.html). To get all payment details, download the attachment and deserialize it.
::


```

File: content/4.examples/query-payment-states.md [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
+  MyPaymentService ..> GevamuFacade
+```
+
+## Payment structure
+
+Structure of the [`Payment`](https://gevamu.github.io/corda-payments-sdk/payments-contracts/com.gevamu.corda.states/-payment/index.html) is as follows:
+
+::notice{info}
+  Pay attention, that full payment instruction is stored as Corda [`Attachment`](https://docs.r3.com/en/api-ref/corda/4.8/open-source/javadoc/net/corda/core/contracts/Attachment.html). So you need to download it and deserialize it to get all payment details.
+::
```
Comment (lingvolab) [inline content/4.examples/query-payment-states.md:24]:
```suggestion
```

File: content/4.examples/query-payment-states.md [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+  }
+
+  MyPaymentService ..> GevamuFacade
+```
+
+## Payment structure
+
+Structure of the [`Payment`](https://gevamu.github.io/corda-payments-sdk/payments-contracts/com.gevamu.corda.states/-payment/index.html) is as follows:
+
+::notice{info}
```
Comment (lingvolab) [inline content/4.examples/query-payment-states.md:22]:
```suggestion
```

File: content/4.examples/query-payment-states.md (hunk 12 above)
Comment (lingvolab) [reply content/4.examples/query-payment-states.md:58]:
```suggestion
::notice{info}
  Note that the full payment instruction is stored as a Corda [`Attachment`](https://docs.r3.com/en/api-ref/corda/4.8/open-source/javadoc/net/corda/core/contracts/Attachment.html). 
::

```

File: content/4.examples/query-payment-states.md (hunk 6 above)
Comment (lingvolab) [reply content/4.examples/query-payment-states.md:66]:
```suggestion
  Gevamu SDK doesn't provide any ready-to-use solution for querying by payments details stored in the attachment. However, you can create additional tables with the information needed for your business needs and use them in vault queries. 
  To get all payment details from the attachment, download and deserialize it.
 
```

**Review by ASitnikov — APPROVED** [review]
Looks good to me

**Pull request merged**